### DISCUSSION TITLE
Is binary '/bin/cpptools' a tool that integrates GDB and LLDB?

### DISCUSSION THREAD

**Discussion by inlann** [microsoft/vscode-cpptools#8934] [Q&A] [answered]
Hello:

The [document](https://github.com/microsoft/vscode-cpptools/blob/main/Documentation/Building%20the%20Extension.md) says I need to download the cpptools extension from the Visual Studio Marketplace and copy its binaries `cpptools` and `cpptools-srv` into my bin folder. May I ask that is binary `/bin/cpptools/` a tool(debugger) that integrates GDB and LLDB?

**Comment by michelleangela** [comment, marked as answer]
The binaries `cpptools` and `cpptools-srv` under the bin folder are the binaries of the Language Server and IntelliSense Engine. They have nothing to do with debugging.

Could you clarify what you mean by "a tool that integrates GDB and LLDB"? Are you trying to build the extension to change something with the debugger?

**Reply by inlann** [reply]
Thanks 😄 
Yes I'm trying to customize the extension to adapt my debugger based on LLDB. Seems like a specific debugConfigurationProvider is enough.
